upgrade/upgrade.go

```diff
@@ -268,6 +270,19 @@ func doResourceUpgrade(namespace string, lhClient *lhclientset.Clientset, kubeCl
 			return err
 		}
 	}
+	if semver.Compare(lhVersionBeforeUpgrade, "v1.7.1") < 0 {
+		logrus.Info("Walking through the resource upgrade path v1.7.0 to v1.7.1")
+		if err := v170to171.UpgradeResources(namespace, lhClient, kubeClient, resourceMaps); err != nil {
+			return err
+		}
+	}
+	// When lhVersionBeforeUpgrade < v1.8.0, it is v1.7.x. The `CheckUpgradePathSupported` method would have failed us out earlier if it was not v1.7.x.
+	if semver.Compare(lhVersionBeforeUpgrade, "v1.8.0") < 0 {
+		logrus.Info("Walking through the resource upgrade path v1.7.x to v1.8.0")
+		if err := v17xto180.UpgradeResources(namespace, lhClient, kubeClient, resourceMaps); err != nil {
+			return err
+		}
+	}
```

Comment:
We can remove this when backporting to v1.7.x

Reply:
manually create the backport PR: https://github.com/longhorn/longhorn-manager/pull/3125